CromIAM/src/main/scala/cromiam/sam/SamClient.scala

```diff
@@ -24,11 +27,24 @@ class SamClient(scheme: String, interface: String, port: Int, log: LoggingAdapte
                                                                                    materializer: ActorMaterializer) extends StatusCheckedSubsystem {
   override val statusUri = uri"$samBaseUri/status"
 
-  def collectionsForUser(user: User): Future[List[Collection]] = {
-    import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
-    import akka.http.scaladsl.unmarshalling.Unmarshal
-    import SamResourceJsonSupport._
+  def isSubmitWhitelisted(user: User): Future[Boolean] = {
+    val request = HttpRequest(
+      method = HttpMethods.GET,
+      uri = samSubmitWhitelistUri,
+      headers = List[HttpHeader](user.authorization)
+    )
 
+    for {
+      response <- Http().singleRequest(request)
+      whitelisted <- response.status match {
+        case StatusCodes.OK => Unmarshal(response.entity).to[String].map(_.toBoolean)
+        case _ => Future.successful(false)
+      }
+      _ = if (!whitelisted) log.error("Submit Access Denied for user {}", user.userId)
```

Comment:
ToL: `user.userId` is information that they gave us, right? Not that we've dug up from SAM for them?

Reply:
Look at how it is being called